### ISSUE TITLE
VS2017?

### ISSUE THREAD

**Issue by hansmbakker** [Microsoft/FFmpegInterop#136]
It seems that BuildFFmpeg.bat wants to use VS2015. Could you please update it to support VS2017?

**Comment by khouzam** [comment]
Yes, the way to detect the tools has changed with VS2017. We are looking at updating the scripts to support VS2017.

**Comment by ryanwinter** [comment]
Here's the updated Buildffmpeg.bat I used for vs 2017
[BuildFFmpeg.zip](https://github.com/Microsoft/FFmpegInterop/files/988778/BuildFFmpeg.zip)

**Comment by farazp** [comment]
Hi, this isn't for VS2017

**Comment by khouzam** [comment]
The code works fine for VS 2017, we just haven't had a chance to fully update the build scripts to support VS 2017. I've got most the necessary changes but need to validate with different install combinations.

**Comment by dantler** [comment]
In the interim, you could try [this](https://github.com/dantler/FFmpegInterop/commit/d42f1ece775856902f7887ee061a674801fdc145).  Worked for me.

**Comment by farazp** [comment]
I try both, but I got error "VS140COMNTOOLS environment variable is not found. Check your Visual Studio 2015 installation" how to fix this?

**Comment by khouzam** [comment]
This is because the scripts haven't been updated for VS2017.

I've been investigating a new package manager that Microsoft is developing and it does make things a lot easier to get started and works great for both VS2015 and VS2017. That's partly why it's taken me longer to get the scripts updated to VS2017.

The project is [vcpkg](https://github.com/microsoft/vcpkg) and it already supports a bunch of different projects with both VS2015 and VS2017. They have already added support for FFmpeg and I've issued a pull request to add the UWP version of FFmpeg.

To try it out is quite simple:
* Clone the repo (for now use my fork until my code is merged upstream)
`git clone https://github.com/khouzam/vcpkg -b ffmpeg-uwp`
* Bootstrap the tool:
`cd vcpkg`
`bootstrap-vcpkg.bat`
*Build the flavors of FFmpeg that you need for example if your want x86, x64 and arm (the -uwp will build for store whereas -windows will build for desktop).
`vcpkg install ffmpeg:x86-uwp ffmpeg:x64-uwp ffmpeg:arm-uwp`
Let this run for a while and you will endup with FFmpeg built and available for use (noticed that msys, and all the tools are automatically installed and configured for you.

Now comes the even more interesting part, the integration with VS.
* From an elevated command prompt run
`vcpkg intergrate install`
This configures VS to automatically reference the packages that you have built and have installed. So any project can now reference the FFmpeg headers without having to modify the include path and manually adding references to the dlls. Building a store app will automatically include the required files in the package.

You can see how this has simplified the projects with the [vcpkg](https://github.com/Microsoft/FFmpegInterop/tree/gillesk/vcpkg) branch that I just updated.

**Comment by touseefbsb** [comment]
@khouzam  hey I just wanted to ask that the steps you defined above if I follow them for my VS 2017, I don't need to clone this repo "github.com/Microsoft/FFmpegInterop" and I don't need to follow other details steps mentioned here "https://trac.ffmpeg.org/wiki/CompilationGuide/WinRT" right?

**Comment by touseefbsb** [comment]
I was able to create all 3 packages for UWP, x86, x64, and arm. I also integrated them with VS according to the steps you mentioned in above comment. Now how can I consume and reference those created packages in my UWP project? Thanks

**Comment by touseefbsb** [comment]
I just wanted to ask why cant we have a nugget package for this repo? So that it gets simpler to integrate it in any uwp project on any machine instead of building it and then integrating it on a machine whenever we want to use it?

**Comment by touseefbsb** [comment]
the reason I asked thisis because on some machines it builds successfully but on some machines it gives many errors, so if there was a straight forward package that would simplify things

**Comment by ahmar20** [comment]
How do you reference these integrated packages in VS 2017 for C#?

**Comment by lukasf** [comment]
@touseefbsb I think there are legal reasons why no compiled builds of ffmpeg are distributed. Some of the codecs might be covered by patents. The authors of ffmpeg "only show you how this could be decoded" but they do not offer any running decoders/encoders. If you compile yourself, then you are the one who acutally built the decoder assembly...
